rfcs/react-components/convergence/upgrade-code-location.md

```diff
@@ -0,0 +1,126 @@
+# RFC: Upgrade code location
+
+---
+
+@GeoffCoxMSFT
+
+## Summary
+
+Partners and customer developers will need some helper code to upgrade from v0 or v8 to v9.
+This will include
+
+- shim components - take v0 or v8 props and render v9 components
+- shim functions - functions to help map between v0/v8 themes, styles, etc. and v9
+- code mod scripts - scripts that update code calling v8 to call v9 components
+- examples - web applications or stories that show before (v0/v8), during (side-by-side), and after (v9)
+
+## Background
+
+We currently have code for upgrade v8 to v9 in the [fluentui-react-upgrade repository](https://github.com/microsoft/fluentui-react-upgrade). This repository is currently accessibly only within the Microsoft organization. This repo also has the upgrade guide for v8 to v9 which is in progress being moved into the v9 storybook documentation.
+
+So far the few partners that have done flighting of v9 components prefer access to the shim source code that they can extract into their own repositories to modify for their specific flighting scenarios.
+They have not yet asked for a package of shims to take a dependency on, but prefer to write their own shims if they are not upgrading to v9 directly.
+They have tended to author their own code-mod scripts.
+
+## Problem statement
+
+Where should upgrade helper code live?
+
+## Detailed Design or Proposal
+
+### Option A: Create upgrade packages in the fluent repo
```

Comment:
Option A seems sound to keep complex dependencies out of react-components. Then we can build out a unique storybook for the upgrades and then use storybook composition to add these stories into our main storybook site

https://www.chromatic.com/docs/composition

Reply:
this option is my personal favorite. keeps the structure clean and the storybook composition is something we have been planning to use for partners anyway.

Reply:
I agree that this is likely the best approach for this type of code.